### PR TITLE
send email upon sign up

### LINKED ISSUE #90: Send email on registration complete
After the user signs up for an event, we should email them a confirmation email. Using the `/mail/register` route, send a user an email after they click register on the signup page. Note that you will have to set some new environment variables in order to get this back end route to work correctly. You can get the environment variables from @vwinstea on Slack.

### PR DESCRIPTION
**Pull request by adellevo**
* email is sent out when volunteer signs up for event
* prob should refactor so that the sendEmail function can be shared between the two components (SignUpForm and WaiverPage) 
--------------------------------
* still works even with this error, but not sure how to fix this
Screen Shot 2022-04-24 at 2 21 25 PM

### CODE REVIEW

File: backend/package-lock.json [hunk 1]
```diff
@@ -4,7 +4,6 @@
   "requires": true,
   "packages": {
     "": {
-      "name": "backend",
```
Comment (vwinstea) [inline backend/package-lock.json]: Any ideas why this ended up getting removed?

File: backend/routes/mail.js [hunk 2]
```diff
@@ -24,16 +20,15 @@ const transporter = nodemailer.createTransport({
...
     transporter.sendMail(
       {
         from: `City Farm SLO ${process.env.EMAIL_USER}`,
         to: user.email,
-        subject: `Registered for ${event.title}`,
+        subject: `Registered for ${event}`,
         html: `<p>Hello ${user.firstName} ${user.lastName},</p>
             <p>You are registered for:</p>
-            <code>${JSON.stringify(event)}</code>`,
+            <code>${event}</code>`,
```
Comment (vwinstea) [inline backend/routes/mail.js:35]: I'll make a task for this for next week so we can add a bit more info to the email.

File: frontend/src/components/UserSignUp/SignUpForm.jsx [hunk 3]
```diff
@@ -234,6 +234,22 @@ function SignUpForm({ selectedEvent, handleModalClose, isAdult, setUser }) {
   // routing
   const history = useHistory();
 
+  const sendEmail = (data) => {
+    fetch("mail/register", {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
```
Comment (vwinstea) [inline frontend/src/components/UserSignUp/SignUpForm.jsx]: I think this is duplicate code from the WaiverPage

**Review by vwinstea — CHANGES_REQUESTED** [review]
Looking good! I tested it out and I did get the email, so that is working well. If you could do some code quality touchups (linting, removing console logs, removing duplicate code), that would be great! Also, idk what exactly is causing the error, but I've never seen `res.clone()` before, so maybe that has something to do with it? Anyways, lmk if you have any questions

File: backend/routes/mail.js (hunk 2 above)
Comment (adellevo) [reply backend/routes/mail.js:35]: Sounds good.

File: frontend/src/components/UserSignUp/SignUpForm.jsx (hunk 3 above)
Comment (adellevo) [reply frontend/src/components/UserSignUp/SignUpForm.jsx]: Yep, just refactored!

File: backend/package-lock.json (hunk 1 above)
Comment (adellevo) [reply backend/package-lock.json]: No idea, but I added it back because that was unintentional.

**Review by vwinstea — APPROVED** [review]
Looks good! Thanks for making those fixes!

**Pull request merged**